Quotient/events/roomjoinrulesevent.h

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: 2025 James Graham <james.h.graham@protonmail.com>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <Quotient/events/stateevent.h>
+
+namespace Quotient
+{
+namespace EventContent {
+Q_NAMESPACE_EXPORT(QUOTIENT_API)
+
+//! \brief Definition of an allow AllowCondition
+//!
+//! \sa https://spec.matrix.org/latest/client-server-api/#mroomjoin_ruless
+struct AllowCondition {
+    QString roomId;
+    QString type;
+};
+
+//! Enum representing the available room join rules
+enum JoinRule {
```

Comment:
Always make sure your enums specify the underlying integer type. The standard [leaves it for implementations to decide](https://en.cppreference.com/w/cpp/language/enum#Unscoped_enumerations), so MSVC uses `int` if the type is omitted while GCC and both flavours of Clang use `unsigned int` - this is why the CI is failing on Windows now. I added the same integer type `enum Membership` uses but if you prefer `uint8_t` I won't mind :)
```suggestion
enum JoinRule : uint16_t {
```